src/sync/app.hpp

```diff
@@ -280,12 +260,23 @@ class App {
     std::string m_app_route;
     std::string m_auth_route;
     uint64_t m_request_timeout_ms;
+
+    void refresh_access_token(std::function<void(Optional<AppError>)> completion_block) const;
```

Comment:
Generally, it would be nice with a high-level doc around how tokens and refreshes work. Perhaps at the top of this file or at `do_authenticated_request`

From my understanding, all REST calls must use `do_authenticated_request` which is responsible for detecting if a token refresh is needed and do that. 

In the past, we attempted to refresh the access token before it expired, but that was mostly due to sync. I guess now that is handled elsewhere, then just reacting to the server response code for REST calls is much simpler and probably doesn't matter much in terms of user experience.

Reply:
I've added docs, I don't think all calls need `do_authenticated_request` for example `log_in` and `register` don't require this.